work_queue/src/work_queue.c

```diff
@@ -3094,13 +3100,17 @@ Used for computing queue capacity below.
 static void add_task_report(struct work_queue *q, struct work_queue_task *t)
 {
 	struct work_queue_task_report *tr;
+	struct work_queue_stats s;
+	work_queue_get_stats(q, &s);
 
 	// Create a new report object and add it to the list.
 	tr = calloc(1, sizeof(struct work_queue_task_report));
 
 	tr->transfer_time = (t->time_when_commit_end - t->time_when_commit_start) + (t->time_when_done - t->time_when_retrieval);
+	debug(D_WQ, "tr->transfer_time %"PRId64" = (%"PRId64" - %"PRId64") + (%"PRId64" - %"PRId64")\n", tr->transfer_time, t->time_when_commit_end, t->time_when_commit_start, t->time_when_done, t->time_when_retrieval);
 	tr->exec_time     = t->time_workers_execute_last;
-
+	debug(D_WQ, "tr->exec_time = %"PRId64"\n", t->time_workers_execute_last);
+	tr->master_time = (s.time_status_msgs - s.prev_time_status_msgs) + (s.time_internal - s.prev_time_internal);
```

Comment:
The above is per task. master_time is global.  Further, they do not correspond to the time the master dedicated to this task, but to the time between calls to add_task_report. 

If this is to be taken as some sort of average of master_time across tasks, then there should better/clearer ways to get it.